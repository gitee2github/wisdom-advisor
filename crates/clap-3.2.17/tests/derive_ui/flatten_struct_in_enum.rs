#[derive(clap::Parser)]
enum Opt {
    #[clap(flatten)]
    Sub(SubCmd),
}

#[derive(clap::Parser)]
struct SubCmd {}

fn main() {}
