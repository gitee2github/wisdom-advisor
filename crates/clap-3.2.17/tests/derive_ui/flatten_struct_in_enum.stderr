error[E0277]: the trait bound `SubCmd: Subcommand` is not satisfied
   --> tests/derive_ui/flatten_struct_in_enum.rs:1:10
    |
1   | #[derive(clap::Parser)]
    |          ^^^^^^^^^^^^ the trait `Subcommand` is not implemented for `SubCmd`
    |
note: required by `has_subcommand`
   --> src/derive.rs
    |
    |     fn has_subcommand(name: &str) -> bool;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: this error originates in the derive macro `clap::Parser` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `SubCmd: Subcommand` is not satisfied
   --> tests/derive_ui/flatten_struct_in_enum.rs:1:10
    |
1   | #[derive(clap::Parser)]
    |          ^^^^^^^^^^^^ the trait `Subcommand` is not implemented for `SubCmd`
    |
note: required by `augment_subcommands`
   --> src/derive.rs
    |
    |     fn augment_subcommands(cmd: Command<'_>) -> Command<'_>;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: this error originates in the derive macro `clap::Parser` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `SubCmd: Subcommand` is not satisfied
   --> tests/derive_ui/flatten_struct_in_enum.rs:1:10
    |
1   | #[derive(clap::Parser)]
    |          ^^^^^^^^^^^^ the trait `Subcommand` is not implemented for `SubCmd`
    |
note: required by `augment_subcommands_for_update`
   --> src/derive.rs
    |
    |     fn augment_subcommands_for_update(cmd: Command<'_>) -> Command<'_>;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: this error originates in the derive macro `clap::Parser` (in Nightly builds, run with -Z macro-backtrace for more info)
